Cubical/Foundations/Univalence.agda

```diff
@@ -149,6 +149,9 @@ univalencePath = ua (compEquiv univalence LiftEquiv)
 uaβ : {A B : Type ℓ} (e : A ≃ B) (x : A) → transport (ua e) x ≡ e .fst x
 uaβ e x = transportRefl (e .fst x)
 
+uaη : ∀ {A B : Type ℓ} → (P : A ≡ B) → ua (pathToEquiv P) ≡ P
+uaη = J (λ _ q → ua (pathToEquiv q) ≡ q) (cong ua pathToEquivRefl ∙ uaIdEquiv)
+
```

Comment:
Is there a simpler definition of this? With a different `ua` we can do like so:

    uaa : (E : A ≃ B) → A ≡ B
    uaa {A = A} {B = B} E i
      = Glue B (λ{ (i = i0) → A , E ; (i = i1) → B , pathToEquiv (λ _ → B) })

    uaη : (P : A ≡ B) → uaa (pathToEquiv P) ≡ P
    uaη {A = A} {B = B} P i j
      = Glue B λ where
          (j = i0) → A , pathToEquiv P
          (i = i1) → P j , pathToEquiv (λ i → P (j ∨ i))
          (j = i1) → B , pathToEquiv (λ i → B)

Reply:
I tried taking your definition and "patching it up" and I ended up with something that looks complicated:

```
uaη : ∀ {A B : Type ℓ} → (P : A ≡ B) → ua (pathToEquiv P) ≡ P
uaη {A = A} {B} P i j =
  (Glue B λ where
      (j = i0) → A , pathToEquiv P
      (i = i1) → P j ,
         equivEq
           (pathToEquiv (λ i → P (j ∨ i)))
           (transp (λ k → P (~ k ∨ j) ≃ B) j (idEquiv B))
           (λ i x → transp (λ _ → B) (j ∨ ~ i) (transp (λ k → P (k ∨ j)) (j ∧ i) x)) j
      (j = i1) → B , idEquiv B)
```

Reply:
That does normalize to about 25 lines instead of 75. I guess I'm not sure how much that sort of thing matters, either.

Reply:
Here's something slightly shorter (both in source and normalized form, it seems):
```
uaη : ∀ {A B : Type ℓ} → (P : A ≡ B) → ua (pathToEquiv P) ≡ P
uaη {A = A} {B} P i j =
  (Glue B λ where
      (j = i0) → A , pathToEquiv P
      (i = i1) → P j ,
         hcomp (λ k → λ {
           (j = i0) → pathToEquiv P;
           (j = i1) → pathToEquivRefl k})
         (pathToEquiv (λ i → P (j ∨ i)))
      (j = i1) → B , idEquiv B)
```

Reply:
By the way, on line 117 there seems to be a verbatim copy of your original proof. It could probably be extracted too by `retEq univalence` or similar.